### PR TITLE
[JENKINS-45948] Bump AHC dependency to final 1.x release

### PR DESCRIPTION
**Pull request by ghost**
@reviewbybees @stephenc @daniel-beck 

https://issues.jenkins-ci.org/browse/JENKINS-45948

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -63,7 +63,7 @@
 
   <properties>
     <jenkins.version>1.580.1</jenkins.version>
-    <java.level>6</java.level>
+    <java.level>7</java.level>
```
Comment (oleg-nenashev) [inline pom.xml]: 🐛 Jenkins core should be bumped to 1.625.3 then

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

File: pom.xml (hunk 1 above)
Comment (ghost) [reply pom.xml]: Knew I forgot something, thanks.

**Review by oleg-nenashev — APPROVED** [review]
:bee:

**Comment by recampbell** [comment]
@rysteboe, have you tested this with affected dependent plugins?

**Comment by ghost** [comment]
@recampbell yes, I have.

**Comment by ghost** [comment]
@reviewbybees done

**Review by stephenc — APPROVED** [review]

**Comment by stephenc** [comment]
@rysteboe do you need me to merge?

**Comment by ghost** [comment]
@stephenc oh apparently I don't anymore!  Yay new permissions.  I think you still need to do the release though?

**Comment by stephenc** [comment]
@rysteboe can you update the wiki with the `1.9.40.0` release (note  `.0` and not `.1` as this is the first release based off `1.9.40` and hence is `1.9.40` async-http-client)

**Comment by ghost** [comment]
@stephenc done

**Pull request merged**